libs/utils/analysis_tool.py

```diff
@@ -0,0 +1,83 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited, Google and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import logging
+import os
+import select
+from trace import Trace
+from conf import LisaLogging
+from devlib.utils.misc import memoized
+
+class AnalysisTool(object):
+    """
+    A base class for LISA analysis command-line tools
+
+    This class intended to be subclassed in order to create a custom
+    analysis tool.
+    """
+
+    @memoized
+    def parse_args(self, parser):
+        if not parser:
+            parser = argparse.ArgumentParser(
+                    description='LISA Analysis tool configuration')
+
+        parser.add_argument('--trace', type=str,
+                default=None,
+                help='Path of the trace file or directory to be used',
+                required=True)
+
+        parser.add_argument('--log-stdout', action='store_true',
+                default=False,
+                help='Output logging to stdout',
+                required=False)
+
+        # Parse command line arguments
+        return parser.parse_args()
+
+    def run_analysis(self):
+	raise NotImplementedError('error')
+
+    def run_output(self):
+        if self.gui:
+            raise NotImplementedError('error')
+        print self.output
+
+    def __init__(self, parser=None, gui=False):
```

Comment:
Will add proper docstrings, sorry for making the review hard. GUI is supposed to be for tools that want to show output in the browser instead of text. The idea is for certain tools to create a report automatically (using ipython notebook generation utilities) and automatically open it in the browser.